### DISCUSSION TITLE
App not installed (AIR SDK 33.1.1.575)

### DISCUSSION THREAD

**Discussion by Kjurchiska** [airsdk/Adobe-Runtime-Support#1094] [Q&A]
I successfully export .aab without any error with IntelliJ and licensed AIR SDK 33.1.1.575, but when I build apks from .aab, I can not install it on the device with bundletool local, when I Open It I receive the message "App not installed". Also, when I export .aab besides .aab file, there is AndroidStudioProject and I Generated a signed bundle with Android Studio, then I build apks for my device and I have the same message.

I was trying to upload to google play console with the newest certificate and then install it, I did that successfully without errors, but when I try to install it on the device from "Internal test" I have the same message "App not installed". 
I tried to erase all test release and upload the newest but didn't help. 
So, I researched and I find out that this can be caused by Corrupted files, Not enough storage, Insufficient system permissions, or Unsigned App. I was trying to change versions many times, but I get the same error. I tried to disable Google Play Protect, but still, get the same error. I tried with "allowing unknown app installation", but it also didn't help. I tried to combine all possible combinations with anes, include/exclude it, even I tried to export build without anes, but still, I had the same issue.

The main problem is that I didn't have any concrete error to know for example which files are corrupted or something like that, I was just trying with all the things that I can find by googling. 
So, Is there any software or support that can help me to figure out why I have that message and how to discover what is exactly the issue why I can not install apk from .aab?

**Comment by ajwfrost** [comment]
Hi

A couple of things I think to check:
1) can you please delete the .apk (or .aab) file, and then re-generate it? There is a problem where changes are made to an app, at the moment the files aren't deleted before being regenerated and due to the way in which they're generated (as a dynamic zip file) it means we can end up with invalid code signatures...
2) if you've got a problem with the bundletool not generating APKs can you check you have the debug keystore available (build/deploy a test application from Android Studio, which should generate it).

thanks

**Reply by Kjurchiska** [reply]
Hi,

1. I always delete the folder where the .aab is generated and then I regenerated.
2. I don't have a problem with bundletool, I successfully generate the apks zip folder, even I tried with "--connected-device" and the apks that were generated were only for my connected device on pc. But when I try to install that apk on my device I receive the message "App not installed". I tried with all apks, base.apk, etc. Also, I tried the same with Google Play Console but still can not install apks.
